app/models/airline.rb

```diff
@@ -1,3 +1,10 @@
 class Airline < ApplicationRecord
    has_many :flights
-end
\ No newline at end of file
+
+   def adult_passengers
+      Passenger.joins(:flights)
+         .where(flights: { airline_id: id })
+         .where('passengers.age >= 18')
+         .distinct
```

Comment:
this works, there is an easier way to do it though by adding more model associations of has_many passengers.